### ISSUE TITLE
Browser Wish List - Tabs and bookmarks are the same thing

### ISSUE THREAD

**Issue by karlcow** [karlcow/otsukare.info#8]
You may comment on [Browser Wish List - Tabs and bookmarks are the same thing ](https://www.otsukare.info/2021/05/10/tabs-bookmarks-ui)- May 10, 2021

**Comment by karlcow** [comment]
Someone sent me a message saying. 

> Hey, I saw your blog post about tabs and bookmarks being the same thing. They aren't though. Tabs save history. You can go backwards and forwards to see how you arrived at a place.


true. good points to articulate
I'll go further in details. tabs is a context with a timeline. bookmark is a context without a timeline (as managed today).
but I probably will need to draw a bit more to explain what I mean by this.

**Comment by jesup** [comment]
Tabs currently beat bookmarks in many cases for me due to 3 things:
1. Spatial relevance - tabs typically have spatial relationships to the tabs near them - opened in a similar timeframe (usually), often related subject-wise (several tabs to articles researching topic A; a set of related bug tabs, etc)
2. Tabs have history - it remembers not only where you are, but also how you got there (go back to to the search or site that sent you there)
3. You don't have to consciously choose a permanent hierarchical position for them, and they automatically go away when you're done with them and close the tab.  

This is why I currently (in this one browser of the 5 devices I use daily) has 6132 tabs (94 open).  I also use bookmarks, but more for stuff I *know* I want to retain permanently.

**Comment by karlcow** [comment]
There is a discussion on linkedin also about it
with https://www.linkedin.com/in/josephcheechang

https://www.linkedin.com/feed/update/urn:li:activity:6797348461374251008?commentUrn=urn%3Ali%3Acomment%3A%28activity%3A6797348461374251008%2C6797462946189463552%29

> Hey Karl! Really like how you described tabs and bookmarks and opened and closed context. I also often see my opened and saved tabs as essentially the same thing (even though they have very different affordances right now). Also wondering what your thoughts are on how we currently handle closed tabs? For example, the built-in browser history interfaces or those tab session add-ons like Session Buddy or OneTab. 

which I replied to with:


> History/Tabs/Bookmarks, I should added this. That's a good point. If we can consider that Tabs are bookmarks with a timeline (bi-directional). The history is just a view, I mention it at the end of my post. 
>
> What I dislike in the current management of the information is that we get these views in complete separate world. Tabs are this collection of top horizontal or vertical list of boxes for contexts. History is a list of timestamps of the closed and opened contexts. And bookmarks is a pin on opened and closed contexts.
> 
> Be for my OS or my mails, I'm mainly using "smart folders" and "smart mailboxes". Aka dynamic views according to search criteria, which mean the same item can belong to many search criteria without duplication.

**Comment by kelimuttu** [comment]
> A bookmark is just a closed context, and a tab is just an opened context

That surely is an interesting notion. However, it requires an assumption that we use the same mental model of using and categorizing bookmark as well as tab. 

For me, the main disconnect from the two (bookmark vs tab) is that I use bookmark to keep things (documentation). In contrary, I use tab to free me from the need of having to check the same page again (operational). 

Still, it will be interesting if we find a way to connect those two. 

And  the tab sketch is surely an interesting piece. It reminds me of one of my favorite add-ons, [Panorama View](https://addons.mozilla.org/en-US/firefox/addon/panorama-view/) - which you should definitely check if you haven't. 😉

**Comment by DavidRegev** [comment]
Hi, Karl. I couldn’t agree with you more!

I’ve been thinking about redesigning the browser for over a decade (including a couple of [guest posts](https://blog.mozilla.org/labs/tag/david-regev/) for Mozilla Labs in 2011 and a [master’s thesis](https://docs.google.com/presentation/d/1CFL-Q8HT4CnHqt8viLHp40mLWW9EjfnOwVafZ7JDVT0/) last year). I believe that you are _absolutely right_ about removing the distinction between tabs and bookmarks. In fact, you can go even further and unify tabs, bookmarks, downloads, and even files all into one type of object.

As you pointed out, tabs have context but bookmarks don’t. We can begin to remove this distinction when we realize that a bookmark is merely one page/URL, while a tab is an ordered set of pages/URLs. Within that set, each page can have other pages as parent and children—context. If we think in terms of pages instead of tabs, then a bookmark becomes merely a page that you want to keep for later. The distinction just disappears!

In my thesis, [I demoed](https://www.youtube.com/watch?v=Ka0rFhpXrXI&t=732s) a proof-of-concept browser where tabs don’t exist. Instead, you have stacks of pages arranged in a zoomable environment. The browser remembers parents and children for all pages. New pages start a new stack, but clicking on a link adds a new page to the stack you’re in. That way, related pages are automatically kept organized together while unrelated threads are kept separate.

In the future, when the real thing is built, you would be able to “bookmark” a page by moving it into a page group/folder that you’ve created and just leaving it there. Since a bookmark is just a page, full-text search is automatically available on all “bookmarked” pages.

Because this browser interface is designed for managing large collections of pages and page stacks, including groupings/folders of these items, it’s not just a browser for the Web, but a _file manager for the Web_. If you download a file, it appears in the browser interface as just another page in the stack. You no longer need to differentiate between opening a link and downloading it.

You could go even further and display all your files in the browser. Not only that, but it can display files you have online as well (and save them offline for you). Now you don’t need to search for that file you downloaded, or any other file, both on your device and in the cloud!

So, to summarize, what I’m saying is that the browser primitive should be the page. A browser built around pages can completely unify the concepts of tabs, bookmarks, downloads, and files, both offline and online. Theoretically, a well-designed browser means you never have to think in terms of those distinct concepts again.

What do you think?

**Comment by karlcow** [comment]
> So, to summarize, what I’m saying is that the browser primitive should be the page. A browser built around pages can completely unify the concepts of tabs, bookmarks, downloads, and files, both offline and online. Theoretically, a well-designed browser means you never have to think in terms of those distinct concepts again.

Yes there are so much more we can do when we change the optics of how we perceive information online. 

Browsers implementers in general became very conservative about changes which are fundamentally changing the habits of people. All these views about informations in different contexts with different UIs are probably the **buildup of features** instead of a comprehensive view of the information we have access to.

I also wonder if the friction to change things come from the code itself in browsers. The current organization is consolidated in code like sediments. Changing completely the way we reorganize information UI requires changing code and **this** has a cost. Experimentations become harder because the layer of data and the layer of visual UI is strongly tied. (Ironically a bit like React and websites).

**Comment by DavidRegev** [comment]
> All these views about informations in different contexts with different UIs are probably the **buildup of features** instead of a comprehensive view of the information we have access to.

Yes, exactly!

What I realized is that we have two interfaces for following links: one interface hides the current page behind the Back button, while the other interface leaves the current page in place while opening the link in a new window/tab. These are two separate ways of dealing with multiple pages, each with its own pros and cons. Whenever you click on a link, you have to decide which interface you want to use.

On top of all that, as you pointed out, we also have the separate interfaces for bookmarks and history. There is a lot of conceptual overlap, leading to confusion, disorder, and time wasted.

Unifying all these disparate interfaces into a better one should be every browser maker’s goal.
